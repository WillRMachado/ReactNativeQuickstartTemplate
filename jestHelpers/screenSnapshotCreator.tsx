import React from 'react';
import renderer from 'react-test-renderer';
import MockedNavigator from './mock/mockedNavigator';
import {act} from 'react-test-renderer';

export default async (Component: React.Component) => {
  //initiated with random number in order to avoid matching a broken snapshot
  let screen: any = Math.random();
  await act(async () => {
    screen = await renderer
      .create(<MockedNavigator component={Component} />)
      .toJSON();
  });

  return screen;
};
