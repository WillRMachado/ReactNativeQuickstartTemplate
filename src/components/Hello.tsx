import React from 'react';
import {View, TouchableOpacity, Text} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {
  decrement,
  increment,
  asyncIncrement,
} from '../store/counter/counter.store';

// import { Container } from './styles';

const Hello: React.FC = () => {
  const sel = useSelector((state: any) => state.store);

  const dispatch = useDispatch();

  return (
    <View>
      <Text>{sel?.counter}</Text>
      <TouchableOpacity
        onPress={() => {
          dispatch(increment(5));
        }}>
        <Text>plus</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => {
          dispatch(asyncIncrement(1));
        }}>
        <Text>plus async</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={{borderWidth: 1}}
        onPress={() => {
          dispatch(decrement());
        }}>
        <Text>minus</Text>
      </TouchableOpacity>
    </View>
  );
};

export default Hello;
