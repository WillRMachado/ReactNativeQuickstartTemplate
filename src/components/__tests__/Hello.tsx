// import React from 'react';
// import MockedNavigator from '../../../jestHelpers/mock/mockedNavigator';
import screenSnapshotCreator from '../../../jestHelpers/screenSnapshotCreator';

import Hello from '../Hello';

it('renders correctly with defaults', async () => {
  const screen = screenSnapshotCreator(Hello);

  expect(screen).toMatchSnapshot();
});
