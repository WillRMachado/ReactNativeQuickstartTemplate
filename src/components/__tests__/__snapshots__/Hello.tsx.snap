// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
Promise {
  "_U": 0,
  "_V": 0,
  "_W": null,
  "_X": null,
}
`;
